### PR TITLE
Lab 6.1 Пофиксил ворнинги тестов и исправил ошибки

### PR DESCRIPTION
**Pull request by miz14**

### CODE REVIEW

**Comment by Raipo7** [comment]
## [Codecov](https://app.codecov.io/gh/Raipo7/TeamN2/pull/28?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None) Report
Patch coverage: **`100.00`**% and no project coverage change.
> Comparison is base [(`ef195ff`)](https://app.codecov.io/gh/Raipo7/TeamN2/commit/ef195ff09fb3502c29e4933d103d5b3a350007b5?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None) 100.00% compared to head [(`e50cc9f`)](https://app.codecov.io/gh/Raipo7/TeamN2/pull/28?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None) 100.00%.

:exclamation: Your organization is not using the GitHub App Integration. As a result you may experience degraded service beginning May 15th. Please [install the Github App Integration](https://github.com/apps/codecov) for your organization. [Read more](https://about.codecov.io/blog/codecov-is-updating-its-github-integration/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None).

<details><summary>Additional details and impacted files</summary>


```diff
@@            Coverage Diff             @@
##              main       #28    +/-   ##
==========================================
  Coverage   100.00%   100.00%            
==========================================
  Files           14        22     +8     
  Lines          315       440   +125     
  Branches        32        34     +2     
==========================================
+ Hits           315       440   +125     
```


| [Impacted Files](https://app.codecov.io/gh/Raipo7/TeamN2/pull/28?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None) | Coverage Δ | |
|---|---|---|
| [SpaceBattle.Lib/ActionCommand.cs](https://app.codecov.io/gh/Raipo7/TeamN2/pull/28?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-U3BhY2VCYXR0bGUuTGliL0FjdGlvbkNvbW1hbmQuY3M=) | `100.00% <100.00%> (ø)` | |
| [SpaceBattle.Lib/Threads/HardStopCommand.cs](https://app.codecov.io/gh/Raipo7/TeamN2/pull/28?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-U3BhY2VCYXR0bGUuTGliL1RocmVhZHMvSGFyZFN0b3BDb21tYW5kLmNz) | `100.00% <100.00%> (ø)` | |
| [SpaceBattle.Lib/Threads/ReceiverAdapter.cs](https://app.codecov.io/gh/Raipo7/TeamN2/pull/28?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-U3BhY2VCYXR0bGUuTGliL1RocmVhZHMvUmVjZWl2ZXJBZGFwdGVyLmNz) | `100.00% <100.00%> (ø)` | |
| [SpaceBattle.Lib/Threads/SendCommand.cs](https://app.codecov.io/gh/Raipo7/TeamN2/pull/28?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-U3BhY2VCYXR0bGUuTGliL1RocmVhZHMvU2VuZENvbW1hbmQuY3M=) | `100.00% <100.00%> (ø)` | |
| [SpaceBattle.Lib/Threads/SenderAdapter.cs](https://app.codecov.io/gh/Raipo7/TeamN2/pull/28?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-U3BhY2VCYXR0bGUuTGliL1RocmVhZHMvU2VuZGVyQWRhcHRlci5jcw==) | `100.00% <100.00%> (ø)` | |
| [SpaceBattle.Lib/Threads/ServerThread.cs](https://app.codecov.io/gh/Raipo7/TeamN2/pull/28?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-U3BhY2VCYXR0bGUuTGliL1RocmVhZHMvU2VydmVyVGhyZWFkLmNz) | `100.00% <100.00%> (ø)` | |
| [SpaceBattle.Lib/Threads/SoftStopCommand.cs](https://app.codecov.io/gh/Raipo7/TeamN2/pull/28?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-U3BhY2VCYXR0bGUuTGliL1RocmVhZHMvU29mdFN0b3BDb21tYW5kLmNz) | `100.00% <100.00%> (ø)` | |
| [SpaceBattle.Lib/Threads/UpdateBehaviourCommand.cs](https://app.codecov.io/gh/Raipo7/TeamN2/pull/28?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-U3BhY2VCYXR0bGUuTGliL1RocmVhZHMvVXBkYXRlQmVoYXZpb3VyQ29tbWFuZC5jcw==) | `100.00% <100.00%> (ø)` | |


</details>

[:umbrella: View full report in Codecov by Sentry](https://app.codecov.io/gh/Raipo7/TeamN2/pull/28?src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None).   
:loudspeaker: Do you have feedback about the report comment? [Let us know in this issue](https://about.codecov.io/codecov-pr-comment-feedback/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None).

File: SpaceBattle.Lib/Threads/SoftStopCommand.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            {
+                thread.HandleCommand();
+            }
+            else
+            {
+                var threadId = IoC.Resolve<string>("Thread.GetIdByThread", thread);
+                IoC.Resolve<ICommand>("Thread.SendCommand", threadId, IoC.Resolve<ICommand>("Thread.HardStopTheThread", threadId)).Execute();
+            }
+        }).Execute();
+        action();
```
Comment (zaryanezrya) [inline SpaceBattle.Lib/Threads/SoftStopCommand.cs]: !

File: SpaceBattle.Lib.Test/Test_ServerThread.cs [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+            return exceptionNotFoundExcepetion.Object;
+        }).Execute();
+        return scope;
+    }
+
+    [Fact]
+    public void Test_CreateAndStartThread_with_HardStopTheThread_AndSoftStopTheThread()
+    {
+
+        var scope = IoCdependency();
```
Comment (zaryanezrya) [inline SpaceBattle.Lib.Test/Test_ServerThread.cs]: !

File: SpaceBattle.Lib.Test/Test_ServerThread.cs [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+        var listcommands = new List<Lib.ICommand>();
+        listcommands.Add(mockCommand.Object);
+        listcommands.Add(mockCommand.Object);
+        listcommands.Add(IoC.Resolve<Lib.ICommand>("Thread.HardStopTheThread", "1"));
+
+        IoC.Resolve<Lib.ICommand>("Thread.SendCommand", "1", listcommands).Execute();
+        System.Threading.Thread.Sleep(100);
+
+        Assert.Equal(UObject.Object.GetProperty("Position"), new Vector(2, 2));
+
```
Comment (zaryanezrya) [inline SpaceBattle.Lib.Test/Test_ServerThread.cs:228]: ?

**Review by zaryanezrya — CHANGES_REQUESTED** [review]

**Comment by miz14** [comment]
Принято на занятии

**Pull request merged**